vpr/src/place/place_constraints.cpp

```diff
@@ -257,3 +258,149 @@ void load_cluster_constraints() {
         }
     }
 }
+
+void mark_fixed_blocks() {
+    auto& cluster_ctx = g_vpr_ctx.clustering();
+    auto& place_ctx = g_vpr_ctx.mutable_placement();
+    auto& floorplanning_ctx = g_vpr_ctx.floorplanning();
+
+    for (auto blk_id : cluster_ctx.clb_nlist.blocks()) {
+        if (!is_cluster_constrained(blk_id)) {
+            continue;
+        }
+        PartitionRegion pr = floorplanning_ctx.cluster_constraints[blk_id];
+        auto block_type = cluster_ctx.clb_nlist.block_type(blk_id);
+        t_pl_loc loc;
+
+        if (is_pr_size_one(pr, block_type, loc)) {
+            //Set block location and grid usage
```

Comment:
Better to give a single higher level comment.  If the block can be placed in exactly one legal (x,y,sub_tile) location, place it now and mark it as fixed.  The current comments don't really add anything useful, as the function called or the data member set makes it clear what that line is doing.